### PR TITLE
Fixed price table and api

### LINKED ISSUE #17: Create fixed price auction
Fixed Price Routes 
Post -> /fixed -> create a fixed auction.. See required fields
GET -> /fixed -> get all current running fixed auctions (get current price based on time)
PUT -> /fixed/:id -> update auction... includes discount price, description, base_price, discount_end
DELETE -> /fixed/:id -> delete auction
POST -> /fixed/:id -> a user buying the auction, set is_finished: True.. Later will create transcation

### PR DESCRIPTION
**Pull request by sltimmins**

### CODE REVIEW

File: backend/routes.js [hunk 1]
```diff
@@ -168,4 +171,176 @@ module.exports = function routes(app, logger) {
...
+  // GET /fixed (get all current fixed options)
+  app.get('/fixed', (req, res) => {
+    pool.getConnection(function (err, connection) {
+      if(err) {
+        // if there is an issue obtaining a connection, release the connection instance and log the error
+        logger.error("Problem obtaining MySQL connection", err);
+        res.status(400).send("Problem obtaining MySQL connection");
+      } else {
+        // if there is no issue obtaining a connection, execute query and release connection
+        const sql = "SELECT * FROM fixed_price";
```
Comment (tmcnitt) [inline backend/routes.js]: should this only be fixed listings that havent been finished?

File: backend/routes.js [hunk 2]
```diff
@@ -168,4 +171,176 @@ module.exports = function routes(app, logger) {
...
+                .status(400)
+                .send({ success: false, msg: "Error deleting auction" });
+            } else {
+              res
+                .status(200)
+                .send({ succes: true, msg: "Auction deleted successfully" })
+            }
+          })
+        }).catch(() => {
+            res.status(400);
```
Comment (tmcnitt) [inline backend/routes.js]: make sure to do res.status(400).end()

File: backend/routes.js (hunk 1 above)
Comment (sltimmins) [reply backend/routes.js]: Good point

File: backend/routes.js [hunk 3]
```diff
@@ -168,4 +171,176 @@ module.exports = function routes(app, logger) {
...
+          connection.release();
+          if(err) {
+            logger.error("Error getting fixed price listings: \n", err);
+            res
+              .status(400)
+              .send({ success: false, msg: "Error getting fixed price listings" });
+          } else {
+            res
+              .status(200)
+              .send({ succes: true, data: rows })
```
Comment (tmcnitt) [inline backend/routes.js]: should we calculate the price based on the discount and time here or should we do that on the frontend? maybe we'll need it for the transcation we generate when the listing is done?

File: backend/routes.js [hunk 4]
```diff
@@ -168,4 +171,176 @@ module.exports = function routes(app, logger) {
...
+    pool.getConnection(function (err, connection) {
+      if(err) {
+        // if there is an issue obtaining a connection, release the connection instance and log the error
+        logger.error("Problem obtaining MySQL connection", err);
+        res.status(400).send("Problem obtaining MySQL connection");
+      } else {
+        // if there is no issue obtaining a connection, execute query and release connection
+        const sql = "SELECT * FROM fixed_price";
+        connection.query(sql, (err, rows) => {
+          connection.release();
```
Comment (tmcnitt) [inline backend/routes.js:225]: here rows are all the listings
we can loop over rows, and set the current price of each to the calculated price based on the discount

rows.forEach((row) => {
   row.current_price = getListingPrice(row.base_price, row.discount_price,row.discount_end)
})

**Pull request merged**